tests/python/frontend/darknet/test_forward.py

```diff
@@ -0,0 +1,437 @@
+"""
+Compile Darknet Models
+=====================
+This article is a test script to test darknet models with NNVM.
+All the required models and libraries will be downloaded from the internet
+by the script.
+"""
+import os
+import requests
+import sys
+import urllib
+import numpy as np
+import tvm
+from tvm.contrib import graph_runtime
+from nnvm import frontend
+from nnvm.testing.darknet import LAYERTYPE
+from nnvm.testing.darknet import __darknetffi__
+import nnvm.compiler
+if sys.version_info >= (3,):
+    import urllib.request as urllib2
+else:
+    import urllib2
+from tvm import relay
+
+def _download(url, path, overwrite=False, sizecompare=False):
+    ''' Download from internet'''
+    if os.path.isfile(path) and not overwrite:
+        if sizecompare:
+            file_size = os.path.getsize(path)
+            res_head = requests.head(url)
+            res_get = requests.get(url, stream=True)
+            if 'Content-Length' not in res_head.headers:
+                res_get = urllib2.urlopen(url)
+            urlfile_size = int(res_get.headers['Content-Length'])
+            if urlfile_size != file_size:
+                print("exist file got corrupted, downloading", path, " file freshly")
+                _download(url, path, True, False)
+                return
+        print('File {} exists, skip.'.format(path))
+        return
+    print('Downloading from url {} to {}'.format(url, path))
+    try:
+        urllib.request.urlretrieve(url, path)
+        print('')
+    except:
+        urllib.urlretrieve(url, path)
+
+DARKNET_LIB = 'libdarknet2.0.so'
+DARKNETLIB_URL = 'https://github.com/siju-samuel/darknet/blob/master/lib/' \
```

Comment:
Can we put this in a centrally located place? there is a TVM data repo somewhere, can we download from Joe's repo?